### ISSUE TITLE
Read about and implement detailed gradient following

### ISSUE THREAD

**Issue by MartinLNG** [MartinLNG/adversarial_training_mps#110]
loss.backward(): How to go through the computational graph and check for existence of gradients.
